nodewatcher/modules/monitor/http/survey/processors.py

```diff
@@ -0,0 +1,156 @@
+import datetime
+import pytz
+
+from django.utils.translation import gettext_noop
+
+from nodewatcher.core.monitor import models as monitor_models, processors as monitor_processors
+from nodewatcher.utils import ipaddr
+from nodewatcher.modules.monitor.sources.http import processors as http_processors
+
+DATASTREAM_SUPPORTED = False
+try:
+    from nodewatcher.modules.monitor.datastream import fields as ds_fields, models as ds_models
+    from nodewatcher.modules.monitor.datastream.pool import pool as ds_pool
+
+    class SurveyInfoStreams(ds_models.RegistryRootStreams):
+        snr = ds_fields.IntegerField(tags={
+            'title': gettext_noop("SNR"),
+            'description': gettext_noop("Node SNR."),
+            'visualization': {
+                'type': 'line',
+                'initial_set': True,
+                'time_downsamplers': ['mean'],
+                'value_downsamplers': ['min', 'mean', 'max'],
+                'minimum': 0.0,
+            },
+        })
+        neighbor_count = ds_fields.IntegerField(tags={
+            'title': gettext_noop("Neighbor count"),
+            'description': gettext_noop("Number of detected access points."),
+            'visualization': {
+                'type': 'line',
+                'initial_set': True,
+                'time_downsamplers': ['mean'],
+                'value_downsamplers': ['min', 'mean', 'max'],
+                'minimum': 0.0,
+            },
+        })
+
+        def get_module_name(self):
+            return 'monitor.http.survey'
+
+    class SurveyInfoStreamsData(object):
+        def __init__(self, node, frequency, channel, snr, neighbors):
+            self.node = node
+            self.frequency = frequency  # either 2 or 5, corresponding to 2.4GHz and 5GHz
+            self.channel = channel
+            self.snr = snr
+            self.neighbors = neighbors
+            self.neighbor_count = len(self.neighbors)
+
+    ds_pool.register(SurveyInfoStreamsData, SurveyInfoStreams)
+
+    DATASTREAM_SUPPORTED = True
+except ImportError:
+    pass
+
+
+class SurveyInfo(monitor_processors.NodeProcessor):
+    """
+    Stores node's reported channel, SNR and neighbor data into the monitoring schema. Will
+    only run if HTTP monitor module has previously fetched data.
+    """
+
+    @monitor_processors.depends_on_context("http", http_processors.HTTPTelemetryContext)
+    def process(self, context, node):
+        """
+        Called for every processed node.
+
+        :param context: Current context
+        :param node: Node that is being processed
+        :return: A (possibly) modified context
+        """
+        # Should I go through the existing clients as well?
+        # existing_clients = {}
+        #for client in node.monitoring.network.clients():
+        #    existing_clients[client.client_id] = client
+
+        version = context.http.get_module_version("core.clients")
```

Comment:
I think that you are using output from the `core.wireless` module, so you should check its version and not `core.clients`.